File: akka-samples/akka-sample-camel/src/main/scala/Transformer.scala
package sample.camel

import se.scalablesolutions.akka.actor.Actor
import se.scalablesolutions.akka.camel.Message

/**
 * @author Martin Krasser
 */
class Transformer(producer: Actor) extends Actor {

  protected def receive = {
    case msg: Message => producer.forward(msg.transformBody[String]("- %s -" format _))
  }

}

File: akka-docs/rst/java/code/docs/camel/sample/route/Transformer.java
package docs.camel.sample.route;
//#CustomRouteExample
import akka.actor.ActorRef;
import akka.actor.UntypedActor;
import akka.camel.CamelMessage;
import akka.dispatch.Mapper;
import akka.japi.Function;

public class Transformer extends UntypedActor {
  private ActorRef producer;

  public Transformer(ActorRef producer) {
    this.producer = producer;
  }

  public void onReceive(Object message) {
    if (message instanceof CamelMessage) {
      // example: transform message body "foo" to "- foo -" and forward result
      // to producer
      CamelMessage camelMessage = (CamelMessage) message;
      CamelMessage transformedMessage =
        camelMessage.mapBody(new Mapper<String, String>(){
          @Override
          public String apply(String body) {
            return String.format("- %s -",body);
          }
        });
      producer.forward(transformedMessage, getContext());
    } else
      unhandled(message);
  }
}
//#CustomRouteExample